python/run-tests.py

```diff
@@ -151,6 +152,61 @@ def parse_opts():
     return opts
 
 
+def _check_dependencies(python_exec, modules_to_test):
+    if "COVERAGE_PROCESS_START" in os.environ:
+        # Make sure if coverage is installed.
+        try:
+            subprocess_check_output(
+                [python_exec, "-c", "import coverage"],
+                stderr=open(os.devnull, 'w'))
+        except:
+            print_red("Coverage is not installed in Python executable '%s' "
+                      "but 'COVERAGE_PROCESS_START' environment variable is set, "
+                      "exiting." % python_exec)
+            sys.exit(-1)
+
+    if pyspark_sql in modules_to_test:
+        # If we should test 'pyspark-sql', it checks if PyArrow and Pandas are installed and
+        # explicitly prints out. See SPARK-23300.
+        try:
+            pyarrow_version = subprocess_check_output(
+                [python_exec, "-c", "import pyarrow; print(pyarrow.__version__)"],
+                universal_newlines=True,
+                stderr=open(os.devnull, 'w')).strip()
+            if LooseVersion(pyarrow_version) >= LooseVersion('0.8.0'):
```

Comment:
I think I can't easily reuse `require_minimum_pandas_version` or `require_minimum_pyarrow_version` since it looks it's not guaranteed to access to our main `pyspark` here. I tries to address the comments at my best. Also updated PR description. Please check the logs above.